### PR TITLE
modification in http collector

### PR DESCRIPTION
**Pull request by mkabischev**

### CODE REVIEW

File: collector-http.go [hunk 1]
```diff
@@ -116,98 +117,91 @@ func (c *HTTPCollector) Close() error {
...
 	}
 
-	req, err := http.NewRequest(
-		"POST",
-		c.url,
-		httpSerialize(sendBatch))
+	data, err := httpSerialize(b)
 	if err != nil {
-		c.logger.Log("err", err.Error())
+		c.logger.Log("err", err)
```
Comment (sbinq) [inline collector-http.go]: so this would be logged twice?

Comment (mkabischev) [reply collector-http.go]: fixed, thx

File: collector-http_test.go [hunk 2]
```diff
@@ -182,38 +182,6 @@ func TestHttpCollector_NonBlockCollect(t *testing.T) {
 
 }
 
-func TestHttpCollector_MaxBatchSize(t *testing.T) {
```
Comment (mkorolyov) [inline collector-http_test.go]: add check that code isn't blocking if http request hangs?

**Review by mkorolyov — CHANGES_REQUESTED** [review]

File: collector-http.go [hunk 3]
```diff
@@ -116,98 +117,90 @@ func (c *HTTPCollector) Close() error {
...
+	defer func() {
+		c.batchPool.Put(b[:0])
+		if err != nil {
+			c.logger.Log("err", err)
+		}
+	}()
 
 	// Do not send an empty batch
-	if len(sendBatch) == 0 {
+	if len(b) == 0 {
```
Comment (hexdigest) [inline collector-http.go:183]: Maybe it's better to put this if statement above the defer?

Comment (mkabischev) [reply collector-http.go:183]: Even if buffer is empty it must be returned to pool.

**Review by mkorolyov — APPROVED** [review]

File: collector-http_test.go (hunk 2 above)
Comment (mkabischev) [reply collector-http_test.go]: There is test for this case: TestHttpCollector_NonBlockCollect, but it did nothing before. I`ve updated it

**Pull request merged**